Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm

```diff
@@ -225,7 +225,9 @@ static bool exceedsRenderTreeSizeSizeThreshold(uint64_t thresholdSize, uint64_t
 
 void WebPageProxy::setDataDetectionResult(DataDetectionResult&& dataDetectionResult)
 {
-    m_dataDetectionResults = WTFMove(dataDetectionResult.results);
+    m_dataDetectionResults = createNSArray(dataDetectionResult.results, [](const RetainPtr<DDScannerResult>& result) {
```

Comment:
This may be outside the scope of this patch, but can `m_dataDetectionResults` also be of type  `Vector<RetainPtr<DDScannerResult>>`?